### DISCUSSION TITLE
Can any one give me example of how to scan barcode from image taken from gallery?

### DISCUSSION THREAD

**Discussion by gaj-programmer** [googlesamples/mlkit#564] [Q&A]
I am getting zero sized barcode list while reading barcode from image taken from gallery and unable to find code sample here.

**Comment by disjustin** [comment]
Android or iOS? Can you provide a logcat, screenshot, and/or the test image that you used? The barcode scanner for the android demo runs and detects barcodes just fine as it currently is.

**Comment by SkGeni** [comment]
How can we use this mlkit in offline mode? Is it possible without any single connection to internet.
